examples/vision/oml_omniglot.py

```diff
@@ -125,9 +145,18 @@ def main(
         print('Meta Valid Error', meta_valid_error / meta_batch_size)
         print('Meta Valid Accuracy', meta_valid_accuracy / meta_batch_size)
 
-        # Average the accumulated gradients and optimize
-        for p in maml.parameters():
-            p.grad.data.mul_(1.0 / meta_batch_size)
+        # sample remember/random set
+        random_batch = tasksets.train.sample()
+        random_data, random_labels = sample_random_set(
+            random_batch, shots, ways, device
+        )
+        # concatenate with trajectory
+        meta_training_test_data = torch.stack(evaluation_data_all_tasks, random_data)
+        meta_training_test_labels = torch.stack(evaluation_labels_all_tasks, random_labels)
+
+        # learn - outer loop
+        loss = loss(learner(meta_training_test_data), meta_training_test_labels)
+        loss.backward()
```

Comment:
Here is the only time `backward` is called, in the outer loop update. Can't be called in the inner loop, since the computational graph points to the original weights - hence the update is done manually through the `maml_update` function.

I'm assuming the entire network is update, including the feature extractor (theta) and the adaptation params (W)